backend/app/api/threads.py

```diff
@@ -28,59 +29,70 @@ class ThreadMessagesPostRequest(BaseModel):
 
 
 @router.get("/")
-async def list_threads(opengpts_user_id: OpengptsUserId) -> List[Thread]:
+async def list_threads(user: AuthedUser) -> List[Thread]:
     """List all threads for the current user."""
-    return await storage.list_threads(opengpts_user_id)
+    return await storage.list_threads(user["user_id"])
 
 
 @router.get("/{tid}/messages")
 async def get_thread_messages(
-    opengpts_user_id: OpengptsUserId,
+    user: AuthedUser,
     tid: ThreadID,
 ):
     """Get all messages for a thread."""
-    return await storage.get_thread_messages(opengpts_user_id, tid)
+    thread = await storage.get_thread(user["user_id"], tid)
```

Comment:
what is this trying to accomplish?

Reply:
Without this, a user could request any thread's messages, including messages from a thread that they don't own. This is because the checkpointer inside `storage.get_thread_messages` just takes a thread_id and grabs all associated messages.

Reply:
I see makes sense, can we do in parallel maybe? To avoid slowing down

Reply:
Sure